OutOfSchool/OutOfSchool.WebApi/Controllers/OrganizationController.cs

```diff
@@ -9,16 +15,133 @@ namespace OutOfSchool.WebApi.Controllers
         {
-            return this.Ok("Hello world");
+            try
+            {
+                return this.Ok(this.organizationService.GetAll());
+            }
+            catch (Exception ex)
+            {
+                return this.BadRequest(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Get organization by id.
+        /// </summary>
+        /// <param name="id">Key in database.</param>
+        /// <returns>Organization element with some id.</returns>
+        [HttpGet("{id}")]
+        public async Task<ActionResult<OrganizationDTO>> GetOrganizationById(long id)
+        {
+            try
+            {
+                OrganizationDTO organizationDTO = await this.organizationService.GetById(id).ConfigureAwait(false);
+                return this.Ok(organizationDTO);
+            }
+            catch (ArgumentException ex)
+            {
+                return this.BadRequest(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Method for creating new organization.
+        /// </summary>
+        /// <param name="organizationDTO">Element which must be added.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        [Authorize(Roles = "organization,admin")]
+        [HttpPost]
+        public async Task<ActionResult<Organization>> CreateOrganization(OrganizationDTO organizationDTO)
+        {
+            if (!this.ModelState.IsValid)
+            {
+                return this.BadRequest(this.ModelState);
+            }
+
+            try
+            {
+                organizationDTO.UserId = Convert.ToInt64(User.FindFirst("sub")?.Value);
+                OrganizationDTO organization = await this.organizationService.Create(organizationDTO).ConfigureAwait(false);       
+                return this.CreatedAtAction(
+                    nameof(this.GetOrganizations),
+                    new { id = organization.Id },
+                    organization);
+            }
+            catch (Exception ex)
+            {
+                return this.BadRequest(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Update info about some organization in database.
+        /// </summary>
+        /// <param name="organizationDTO">Entity.</param>
+        /// <returns>Organization's key.</returns>
+        [Authorize(Roles = "organization,admin")]
+        [HttpPut]
+        public async Task<ActionResult> Update(OrganizationDTO organizationDTO)
+        {
+            if (organizationDTO == null)
+            {
+                return this.BadRequest("Entity was null.");
+            }
+
+            if (!this.ModelState.IsValid)
+            {
+                return this.BadRequest(this.ModelState);
+            }
+
+            try
+            {
+                this.organizationService.Update(organizationDTO);
```

Comment:
updated organization can be returned from the Update method, there will be no need for a second call to the service